src/test/storage/encoded_segment_test.cpp

```diff
@@ -271,4 +285,55 @@ TEST_P(EncodedSegmentTest, SequentiallyReadEmptyIntSegment) {
   });
 }
 
+TEST_F(EncodedSegmentTest, SegmentReencoding) {
+  auto check_segment_equality = [&](const std::shared_ptr<ValueSegment<int32_t>> verification_segment,
+                                    const std::shared_ptr<BaseEncodedSegment> test_segment) {
+    EXPECT_EQ(verification_segment->size(), test_segment->size());
+    resolve_encoded_segment_type<int32_t>(*test_segment, [&](const auto& encoded_segment) {
+      auto value_segment_iterable = create_iterable_from_segment(*verification_segment);
+      auto encoded_segment_iterable = create_iterable_from_segment(encoded_segment);
+
+      value_segment_iterable.with_iterators([&](auto value_segment_it, auto value_segment_end) {
+        encoded_segment_iterable.with_iterators([&](auto encoded_segment_it, auto encoded_segment_end) {
+          for (; encoded_segment_it != encoded_segment_end; ++encoded_segment_it, ++value_segment_it) {
+            EXPECT_EQ(value_segment_it->is_null(), encoded_segment_it->is_null());
+
+            if (!value_segment_it->is_null()) {
+              EXPECT_EQ(value_segment_it->value(), encoded_segment_it->value());
```

Comment:
Simply add the segments to tables and check for their equality?

Reply:
If I remember correctly, table equality checks are rather expensive.

Reply:
just use check_table_equals, it produces much more readable test failures.

Reply:
Haven't found a check for segment equality. So I need to construct to tables, append a chunk with that segment and then check for equality. Right?

Reply:
Yes, but a function `check_segments_equal` would be the same effort while being more reusable.

Reply:
What do you think about #e71342f? What bothers me a little bit is that the result prints "Tables not equal" in case of a failure. But I'd like to avoid bloating up the .cpp file just to have a special version of the comparison method for segments.

Reply:
No # in front of commits. e71342f (so that I have something to click on)

Reply:
That’s the general idea, yes. Just two things: First, you are ignore TypeCmpMode by requiring the types to be the same in any case (have a look at how that parameter is used later). Second, you would store ReferenceSegments in a data table.

Reply:
Changed both issues (there is a "full version" now and two specializations, similar to `EXPECT_TABLE_EQ`). Still the "issue" with "Tables not equal".

Reply:
@mrzzzrm : I'd like to close most issues here. Apart from the compiler issue (`::opossum::`) the issue with the `check_segments_equal` remains. When it fails, it prints something like "Tables are not equal". Any ideas or is it not worth changing it?